### PR TITLE
STRATCONN-2984 add some performance improvements to generateFile

### PR DESCRIPTION
**Pull request by rhall-twilio**
Optimization improvements to the LiveRamp destination code

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [X] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [X] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [X] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Review by peterdemartini — APPROVED** [review]

File: packages/destination-actions/src/destinations/liveramp-audiences/operations.ts [hunk 1]
```diff
@@ -57,7 +60,20 @@ function generateFile(payloads: Payload[]) {
...
+function createHashOrCache() {
+  const cache = new Map<string, string>()
+  return function hashOrCache(value: string): string {
+    if (!cache.has(value)) {
+      cache.set(value, hash(value))
+    }
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return cache.get(value)!
+  }
 }
```
Comment (peterdemartini) [inline packages/destination-actions/src/destinations/liveramp-audiences/operations.ts:77]: After thinking about this change, unless there are duplicate identifiers in a batch this won't make much a difference. We should track how much it is used to ensure we get any benefit.

Comment (rhall-twilio) [reply packages/destination-actions/src/destinations/liveramp-audiences/operations.ts:77]: makes sense, should we hold off on the merge?

Comment (peterdemartini) [reply packages/destination-actions/src/destinations/liveramp-audiences/operations.ts:77]: Hmm.... would be nice to measure it. How easy is it to add metrics?

**Review by marinhero — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @rhall-twilio this PR has been deployed.

**Pull request merged**